provider/generic_data_source.go

```diff
@@ -27,15 +33,40 @@ func (dataSourceType DataSourceType[T, PT]) read(ctx context.Context, d *schema.
 	if f, hasFilters := d.Get("filters").([]any); hasFilters && len(f) > 0 {
 		filters = f[0].(map[string]any)
 	}
-	values, err := dataSourceType.convert(client).GetAll(filters)
-	if err != nil {
-		return diag.FromErr(err)
+	var genericClient GenericClient[T, PT] = dataSourceType.convert(client)
+	var err error
+	if genericClient.IsUnique == true {
+		value, err := genericClient.GetUnique()
```

Comment:
maybe you could extract the logic of the `isUnique` into another function to make it easier to read